app/classes/observation_report/darwin/images.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module ObservationReport
+  module Darwin
+    # Darwin Core Observations format.
+    class Images < ObservationReport::CSV
+      attr_accessor :observations, :query, :tables
+
+      self.separator = "\t"
+
+      def initialize(args)
+        super(args)
+        self.observations = args[:observations]
+        initialize_query
+      end
+
+      def initialize_query
+        add_joins
+        add_project
+      end
+
+      def add_joins
+        self.tables = {
+          images: Image.arel_table,
+          images_observations: Arel::Table.new(:images_observations),
+          observations: Observation.arel_table,
+          users: User.arel_table,
+          licenses: License.arel_table
+        }
+        self.query = tables[:images]
+        join_table(:images_observations, :image_id, attribute(:images, :id))
+        io_attribute = attribute(:images_observations, :observation_id)
+        join_table(:observations, :id, io_attribute)
+        join_table(:users, :id, attribute(:images, :user_id))
+        join_table(:licenses, :id, attribute(:images, :license_id))
+      end
+
+      def join_table(join_name, join_field, attribute)
+        table = tables[join_name]
+        join_attribute = table[join_field]
+        self.query = query.join(table).on(join_attribute.eq(attribute))
+      end
+
+      def add_project
+        query.project(attribute(:images, :id),
+                      attribute(:observations, :id).as("obs_id"),
```

Comment:
Can you change this to the following?

    attribute(:images_observations, :observation_id).as("obs_id")

Reply:
Yep.  Of course I started from observation ids which is why it didn't occur to me to let it go, but it seems better.  It would be interesting to see if MySQL 8 is smart enough that it doesn't really make any difference.